sql/core/src/main/scala/org/apache/spark/sql/jdbc/README.md

```diff
@@ -0,0 +1,81 @@
+---
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+# JDBC Connection Provider Handling In Spark
+
+This document aims to explain and demystify JDBC connection providers as they are used by Spark
+and the usage or custom provider implementation is not obvious.
+
+## What are JDBC connection providers and why use them?
+
+JDBC connection providers (CPs from now on) are making JDBC connections initiated by JDBC sources.
+When a Spark source initiates JDBC connection it looks for a CP which supports the included driver,
+the user just need to provide the `keytab` location and the `principal`. The `keytab` file must exist
```

Comment:
Looks like the below content doesn't match "why use them?" - below content should go to the another chapter, like "How to configure security on JDBC connection providers?".

If you'd like to say the reason to use JDBC CP is that it's easy to configure security on kerberos authentication, it seems better we can simply say so, and separate the details on another chapter.

Reply:
Restructured this part.